homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -1,291 +1,371 @@
-import socket
+"""Demo implementation of the media player."""
+from __future__ import annotations
 
-import pymusiccast
-import voluptuous as vol
+from typing import Callable
 
-from homeassistant.components.media_player import PLATFORM_SCHEMA, MediaPlayerEntity
+from homeassistant.components.media_player import MediaPlayerEntity
 from homeassistant.components.media_player.const import (
-    MEDIA_TYPE_MUSIC,
+    REPEAT_MODE_ALL,
+    REPEAT_MODE_OFF,
+    REPEAT_MODE_ONE,
+    SUPPORT_CLEAR_PLAYLIST,
     SUPPORT_NEXT_TRACK,
     SUPPORT_PAUSE,
     SUPPORT_PLAY,
     SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_REPEAT_SET,
+    SUPPORT_SELECT_SOUND_MODE,
     SUPPORT_SELECT_SOURCE,
+    SUPPORT_SHUFFLE_SET,
     SUPPORT_STOP,
     SUPPORT_TURN_OFF,
     SUPPORT_TURN_ON,
     SUPPORT_VOLUME_MUTE,
     SUPPORT_VOLUME_SET,
 )
-from homeassistant.const import (
-    CONF_HOST,
-    CONF_PORT,
-    STATE_IDLE,
-    STATE_ON,
-    STATE_PAUSED,
-    STATE_PLAYING,
-    STATE_UNKNOWN,
-)
-import homeassistant.helpers.config_validation as cv
-import homeassistant.util.dt as dt_util
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
 
-_LOGGER = logging.getLogger(__name__)
+from . import MusicCastDataUpdateCoordinator, MusicCastDeviceEntity
+from .const import DEFAULT_ZONE, DOMAIN, REPEAT_MODE_MAPPING
 
-SUPPORTED_FEATURES = (
-    SUPPORT_PLAY
-    | SUPPORT_PAUSE
-    | SUPPORT_STOP
-    | SUPPORT_PREVIOUS_TRACK
-    | SUPPORT_NEXT_TRACK
-    | SUPPORT_TURN_ON
-    | SUPPORT_TURN_OFF
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[list[Entity], bool], None],
+) -> None:
+    """Set up MusicCast sensor based on a config entry."""
+    coordinator: MusicCastDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    name = coordinator.data.network_name
+
+    media_players: list[Entity] = []
+
+    for zone in coordinator.data.zones:
+        zone_name = name if zone == DEFAULT_ZONE else f"{name} {zone}"
+
+        media_players.append(
+            MusicCastMediaPlayer(zone, zone_name, entry.entry_id, coordinator)
+        )
+
+    async_add_entities(media_players, True)
+
+
+MUSIC_PLAYER_SUPPORT = (
+    SUPPORT_PAUSE
     | SUPPORT_VOLUME_SET
     | SUPPORT_VOLUME_MUTE
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_CLEAR_PLAYLIST
+    | SUPPORT_PLAY
+    | SUPPORT_SHUFFLE_SET
+    | SUPPORT_REPEAT_SET
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_SELECT_SOUND_MODE
     | SUPPORT_SELECT_SOURCE
+    | SUPPORT_STOP
 )
 
-KNOWN_HOSTS_KEY = "data_yamaha_musiccast"
-INTERVAL_SECONDS = "interval_seconds"
 
-DEFAULT_PORT = 5005
-DEFAULT_INTERVAL = 480
+class MusicCastMediaPlayer(MediaPlayerEntity, MusicCastDeviceEntity):
```

Comment:
It's safer to start the with integration common entity class in the MRO.
```suggestion
class MusicCastMediaPlayer(MusicCastDeviceEntity, MediaPlayerEntity):
```